⬆️(core) upgrade ansible to 2.7

jmaupetit opened the pull request:
## Purpose

Now that we are running python 3, it's time to also update Arnold's core engine to Ansible 2.7. 

## Proposal

- [x] upgrade to Ansible 2.7.7
- [x] upgrade to openshift 0.8.4
- [x] renamed `openshift_raw` module to `k8s`
- [x] use `k8s_facts` module instead of the `k8s` lookup

jmaupetit committed:
⬆️(core) upgrade ansible to 2.7

Now that we are running python 3, it's time to also update Arnold's core
engine to ansible 2.7. We have also upgraded openshift to its latest
release.

Notable change: the `openshift_raw` module is now deprecated and has
been renamed `k8s`, we've adapted our tasks and playbooks consequently.

jmaupetit commented:
Nota bene: this is a draft work for curiosity. I just want to know what fails after switching to the new `k8s` module.

jmaupetit committed:
♻️(scripts) improve docker run wrappers

We've fixed and improved our Docker run wrapper by adding rock-solid
command line parsing with getopts. The environment file overriding
mechanism is now fully functionnal.

We've also added (ci)-ansible-playbook wrappers with appropriate
defaults to ease scripts maintenance.

jmaupetit committed:
🐛(edxapp) fix MySQL pod secret

A recent update in centos/mysql-57-centos7:5.7 docker image makes
MYSQL_ROOT_PASSWORD and MYSQL_USER/MYSQL_PASSWORD mutually exclusive.
Defining both induces a container crash loop.

jmaupetit committed:
♻️(core) refactor app deletion

Since Ansible 2.7 upgrade, app deletion (during blue/green switch or
next stack deployment) was partially working. We've decided to refactor
this feature thanks to the k8s_facts module and a less convoluted
approach. We've added a new step in the CI to test it, and also
refactored CI scripts to make them more generic.

lunika commented on .circleci/config.yml:
```diff
@@ -400,9 +400,55 @@ jobs:
+            bin/ci-ansible-playbook deploy.yml "richie"
+            # Test service re-deployed with the next route
+            bin/ci-test-service "richie" "Installation successful!" "/" "next"
+            # Test that only one stack has been deployed
```
I think this comment is not needed

jmaupetit replied:
Oops

lunika commented on bin/_config.sh:
```diff
@@ -42,10 +42,13 @@ function _set_openshift_env() {
-#   COMMAND: command to be run in Arnold container (_e.g._ ansible-playbook deploy.yml)
+#   OPTIONS:
+#     -e    file to set environment variables (_e.g._ env.d/production)
+#     -m    mount sources as a volume
```
You should provide an example. I have no idea how to use this option without viewing the code

lunika commented on bin/_config.sh:
```diff
@@ -42,10 +42,13 @@ function _set_openshift_env() {
+#   OPTIONS:
+#     -e    file to set environment variables (_e.g._ env.d/production)
+#     -m    mount sources as a volume
+#     -t    allocate a pseudo-TTY while running command
```
Same here, provide an example.

lunika commented on bin/_config.sh:
```diff
@@ -42,10 +42,13 @@ function _set_openshift_env() {
+#     -e    file to set environment variables (_e.g._ env.d/production)
+#     -m    mount sources as a volume
+#     -t    allocate a pseudo-TTY while running command
+#     -v    verbose mode
```
Same here, provide an example. Should I use `-vv` or `-v=2` ?

lunika replied:
Sorry I thought this option was the ansible verbosity.

lunika commented on create_redirect.yml:
```diff
@@ -49,28 +49,21 @@
-              )
-            ] | flatten
-          }}
+      k8s_facts:
```
"coeur avec les doigts"

lunika commented on delete_project.yml:
```diff
@@ -15,9 +15,18 @@
 
     - import_tasks: tasks/set_vars.yml
 
+    - name: Check if project exists
```
Do you think this task is really needed ? It fails now if the project doesn't exist ?

jmaupetit replied:
I should re-test it but yes: it failed during my first tests :cry:

jmaupetit replied:
You are right. Maybe there was a delay before deletion and I thought nothing happens. BTW this is now fixed.

lunika commented:
I have an issue when I want to deploy marsha. pods, jobs end in error because the env variable `DJANGO_SENTRY_DSN` does not have a real dsn (a real url I think). This is error is not visible in `CI` env because the secret has `DJANGO_SENTRY_DSN` defined. I think the line 19 of file `apps/marsha/templates/app/secret.yml.j2` should be:

```
DJANGO_SENTRY_DSN: "{{ MARSHA_VAULT.DJANGO_SENTRY_DSN | default('') | b64encode }}"
```

jmaupetit commented:
Can you declare an issue for Marsha's secret bug please?

I've addressed all of your comments in separated fixup commits. A quick review would be wonderful to make a new release :pray:

lunika commented:
> Can you declare an issue for Marsha's secret bug please?

Yes I can.

lunika reviewed (approved):
rebase and merge :+1:
